src/middleware/MiddlewareFactory.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module MiddlewareFactory
+ */
+
+import { AuthenticationProvider } from "../IAuthenticationProvider";
+
+import { AuthenticationHandler } from "./AuthenticationHandler";
+import { HTTPMessageHandler } from "./HTTPMessageHandler";
+import { Middleware } from "./IMiddleware";
+import { RedirectHandlerOptions } from "./options/RedirectHandlerOptions";
+import { RetryHandlerOptions } from "./options/RetryHandlerOptions";
+import { RedirectHandler } from "./RedirectHandler";
+import { RetryHandler } from "./RetryHandler";
+import { TelemetryHandler } from "./TelemetryHandler";
+
+/**
+ * @private
+ * To check whether the environment is node or not
+ * @returns A boolean representing the environment is node or not
+ */
+const isNodeEnvironment = (): boolean => {
+	return new Function("try {return this === global;}catch(e){return false;}")(); // tslint:disable-line: function-constructor
+};
+
+/**
+ * @class
+ * Class containing function(s) related to the middleware pipelines.
+ */
+export class MiddlewareFactory {
+	/**
+	 * @public
+	 * @static
+	 * Returns the default middleware chain an array with the  middleware handlers
+	 * @param {AuthenticationProvider} authProvider - The authentication provider instance
+	 * @returns an array of the middleware handlers of the default middleware chain
+	 */
+	public static getDefaultMiddlewareChain(authProvider: AuthenticationProvider): Middleware[] {
+		const middleware: Middleware[] = new Array();
+		const authenticationHandler = new AuthenticationHandler(authProvider);
+		const retryHandler = new RetryHandler(new RetryHandlerOptions());
+		const telemetryHandler = new TelemetryHandler();
+		const httpMessageHandler = new HTTPMessageHandler();
+
+		middleware.push(authenticationHandler);
+		middleware.push(retryHandler);
+		if (isNodeEnvironment()) {
```

Comment:
Why do we not follow redirects in the browser? What ever the reason is, we'll need to do something similar for the .NET client in case of Blazor.

Reply:
This is what I learnt from @finsharp 
" Turns out that Browsers will automatically follow redirects in XHR requests, no matter the conditions. Browsers ALWAYS (as odd as it is) follow redirect, making it unnecessary to have the redirect handler registered"